server/src-lib/Hasura/RQL/Types/ScheduledTrigger.hs

```diff
@@ -0,0 +1,126 @@
+-- | These are types for Scheduled Trigger definition; see "Hasura.Eventing.ScheduledTrigger"
+module Hasura.RQL.Types.ScheduledTrigger
+  ( ScheduleType(..)
+  , ScheduledTriggerName(..)
+  , ScheduledEventId(..)
+  , CreateScheduledTrigger(..)
+  , CreateScheduledEvent(..)
+  , RetryConfST(..)
+  , formatTime'
+  , defaultRetryConfST
+  ) where
+
+import           Data.Aeson
+import           Data.Aeson.Casing
+import           Data.Aeson.TH
+import           Data.Time.Clock
+import           Data.Time.Clock.Units
+import           Data.Time.Format.ISO8601
+import           Hasura.Incremental
+import           Hasura.Prelude
+import           System.Cron.Types
+
+import qualified Data.Aeson                    as J
+import qualified Data.Text                     as T
+import qualified Hasura.RQL.Types.EventTrigger as ET
+
+data RetryConfST
+  = RetryConfST
+  { rcstNumRetries  :: !Int
+  , rcstIntervalSec :: !DiffTime
+  , rcstTimeoutSec  :: !DiffTime
+  , rcstTolerance   :: !DiffTime
+  } deriving (Show, Eq, Generic)
+
+instance NFData RetryConfST
+instance Cacheable RetryConfST
+
+$(deriveJSON (aesonDrop 2 snakeCase){omitNothingFields=True} ''RetryConfST)
+
+defaultRetryConfST :: RetryConfST
+defaultRetryConfST =
+  RetryConfST
+  { rcstNumRetries = 0
+  , rcstIntervalSec = seconds 10
+  , rcstTimeoutSec = seconds 60
+  , rcstTolerance = hours 6
+  }
+
+data ScheduleType = Cron CronSchedule | AdHoc (Maybe UTCTime)
```

Comment:
After discussing with the rest of the team, there's a consensus on keeping the current terminology for `ScheduleType` . 

I think the rational of having an optional timestamp with the `Adhoc` constructor is to create a scheduled event immediately instead of making another API call to `create_scheduled_event`. It's just a matter of convenience. The user can of-course call `create_scheduled_event` after creating the scheduled trigger.